### PR TITLE
 FAS: Routing slip linkage(How its done) | auto complete

### PR DESCRIPTION
**Pull request by shabeeb-aot**
*Issue #:* /bcgov/entity#8530

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/ViewRoutingSlip/RoutingSlipAutoComplete.vue [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      number,
+      numberRules,
+      searchRS,
+      errorMessage,
+      autoCompleteRoutingSlips,
+      searchRoutingSlip,
+      isLoading,
+      search,
+      hideNoData
+    } = useRoutingSlipAutoComplete(props, context)
```
Comment (karthik-aot) [inline src/components/ViewRoutingSlip/RoutingSlipAutoComplete.vue:82]: we dont have any props in this component

File: src/composables/ViewRoutingSlip/useRoutingSlipAutoComplete.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      errorMessage.value = 'Please enter a routing slip - unique ID'
+    } else {
+      errorMessage.value = ''
+    }
+  }
+
+  async function searchRoutingSlip () {
+    isLoading.value = true
+
+    // Add debouncing here
```
Comment (karthik-aot) [inline src/composables/ViewRoutingSlip/useRoutingSlipAutoComplete.ts:39]: try..catch block :-P

**Review by karthik-aot — APPROVED** [review]

**Review by pwei1018 — APPROVED** [review]

File: src/services/routingSlip.services.ts [hunk 3]
```diff
@@ -37,4 +38,13 @@ export default class RoutingSlip {
       `${ConfigHelper.getFasAPIURL()}/routing-slips/queries`, searchParams, { showGlobalLoader: showGlobalLoader }
     )
   }
+
+  public static async saveLinkRoutingSlip (
+    LinkRoutingSlipPrams: LinkRoutingSlipPrams,
+    showGlobalLoader: boolean = false
+  ): Promise<AxiosResponse> {
+    return axios.post(
+      `${ConfigHelper.getFasAPIURL()}/link`, LinkRoutingSlipPrams, { showGlobalLoader: showGlobalLoader }
```
Comment (sumesh-aot) [inline src/services/routingSlip.services.ts:47]: please check endpoint, @saravanpa-aot can confirm

File: src/store/modules/routingSlip.ts [hunk 4]
```diff
@@ -185,8 +195,14 @@ export default class RoutingSlipModule extends VuexModule {
     // formatting as per API
     if (searchRoutingSlipParams.dateFilter) {
       searchRoutingSlipParams.dateFilter = {
-        startDate: CommonUtils.formatDisplayDate(searchRoutingSlipParams.dateFilter[0], 'MM/DD/YYYY'),
-        endDate: CommonUtils.formatDisplayDate(searchRoutingSlipParams.dateFilter[1], 'MM/DD/YYYY')
+        startDate: CommonUtils.formatDisplayDate(
+          searchRoutingSlipParams.dateFilter[0],
```
Comment (sumesh-aot) [inline src/store/modules/routingSlip.ts:199]: this may cause conflict with my PR as date format is changed now

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**